Add SDL3, SDL_shadercross and Dear ImGui

LucasSaragosa opened the pull request.

LucasSaragosa committed:
initial ignore update

LucasSaragosa committed:
Merge branch 'main' into renderer-sdl3

LucasSaragosa committed:
SDL3 integration

dylib rename, dll,todo linux. SO

LucasSaragosa committed:
linux lib

LucasSaragosa committed:
renderer files, linear heap

LucasSaragosa committed:
initial render api

LucasSaragosa committed:
purple

LucasSaragosa committed:
Update Math.cpp

LucasSaragosa committed:
Update Math.hpp

iMrShadow committed:
Remove SDL3 library

iMrShadow committed:
Add SDL3 as a submodule

iMrShadow committed:
Add and link SDL3 in the project

iMrShadow committed:
Add CrossShader as a submodule

iMrShadow committed:
Remove CrossShader submodule

The latest build is broken and the project hasn't received a commit for 3 years.

iMrShadow committed:
Add SDL_shadercross as a submodule

iMrShadow committed:
Update cross-platform workflow to include submodules

iMrShadow committed:
Fix a typo

iMrShadow committed:
Update cross-platform workflow to setup X11 and Wayland dependencies

iMrShadow committed:
Update cross-platform workflow: change Windows environment to experimental windows-2025

iMrShadow committed:
Update cross-platform workflow: add additional Linux dependencies

iMrShadow committed:
Update SDL_shadercross submodule: fix find_package() error

iMrShadow committed:
CMake: Add and link SDL3_shadercross to the project, minor tweaks

iMrShadow committed:
Update cross-platform workflow: minor fix to cloning submodules

iMrShadow committed:
CMake: Unlink SDL_shadercross

iMrShadow commented:
We're postponing adding SDL_shadercross support due to several reasons:
- We currently need only one shader to display models.
- Very long compile times due to SPIR-V and DirectXShaderCompiler.
- CI complications - missing dependencies.
- Having alternatives like pre-built binaries and/or usage of the CLI version.

LucasSaragosa committed:
Render

Render Basic API

LucasSaragosa committed:
samplers,buffers,textures

Render Updates

LucasSaragosa committed:
fix

LucasSaragosa committed:
fix multithread & app context

iMrShadow committed:
Remove old SDL_shadercross files

LucasSaragosa committed:
renderer up to standard

shader parameters, bug fixes, shader format, scene

LucasSaragosa committed:
Merge branch 'renderer-sdl3' of https://github.com/Telltale-Modding-Group/Telltale-Editor into renderer-sdl3

LucasSaragosa committed:
Revert "Merge branch 'renderer-sdl3' of https://github.com/Telltale-Modding-Group/Telltale-Editor into renderer-sdl3"

This reverts commit e0d2c2c48870a7231d6f017eed360d7d36ece0da, reversing
changes made to f2ce953cd45e125f81abf8b4d5f08ee266ecf040.

LucasSaragosa committed:
common formats & renderer

general updates, bug fixes, renderer progress, common formats

LucasSaragosa committed:
rem shadercross

LucasSaragosa committed:
d3dmesh BN100 (mostly), render system updates

LucasSaragosa committed:
bug

LucasSaragosa committed:
more bugs

LucasSaragosa committed:
renderer, bug fixes

LucasSaragosa committed:
tab

LucasSaragosa committed:
Resource System & Ptr rename

LucasSaragosa committed:
bugs + find resources!

LucasSaragosa committed:
FIX ALL TABS

LucasSaragosa committed:
tabs again

LucasSaragosa committed:
T A B

LucasSaragosa committed:
resource bugs + tests

LucasSaragosa committed:
Legacy Games resource system support

LucasSaragosa committed:
tabs & textures!

LucasSaragosa committed:
ISO mount file support

LucasSaragosa committed:
PS2-PK2/ISO/SYMMAP/EXTRACT/BMS3-METASTREAM

PS2 PK2 archives, Symmap files for symbol maps, extraction of files into tasks, BMS3 meta streams.

LucasSaragosa committed:
filesystem

LucasSaragosa committed:
STUPID CXXSTD

LucasSaragosa committed:
version fix

LucasSaragosa committed:
dialog/wbox/rules/audio serialisers

LucasSaragosa committed:
meta bugs, all bone formats

LucasSaragosa committed:
ISO bugs, texture bugs
